python/tvm/relay/frontend/tflite.py

```diff
@@ -1494,6 +1499,112 @@ def convert_transpose_conv(self, op):
+        if inputs[1].qnn_params:
+            cls_pred = _qnn.op.dequantize(data=cls_pred,
+                                          input_scale=inputs[1].qnn_params['scale'],
+                                          input_zero_point=inputs[1].qnn_params['zero_point'])
+        if inputs[2].qnn_params:
+            anchor_expr = _qnn.op.dequantize(data=anchor_expr,
+                                             input_scale=inputs[2].qnn_params['scale'],
+                                             input_zero_point=inputs[2].qnn_params['zero_point'])
+
+        # reshape the cls_pred and loc_prob tensors so
+        # they can be consumed by multibox_transform_loc
+        cls_pred = _op.transpose(cls_pred, [0, 2, 1])
+        # loc_prob coords are in yxhw format
+        # need to convert to xywh
+        loc_coords = _op.split(loc_prob, 4, axis=2)
+        loc_prob = _op.concatenate(
+            [loc_coords[1], loc_coords[0], loc_coords[3], loc_coords[2]], axis=2
+        )
+        loc_prob = _op.reshape(loc_prob, [1, anchor_boxes*4])
+
+        # anchor coords are in yxhw format
+        # need to convert to ltrb
+        anchor_coords = _op.split(anchor_expr, 4, axis=1)
+        anchor_y = anchor_coords[0]
+        anchor_x = anchor_coords[1]
+        anchor_h = anchor_coords[2]
+        anchor_w = anchor_coords[3]
+        plus_half = _expr.const(0.5, dtype='float32')
+        minus_half = _expr.const(-0.5, dtype='float32')
+        anchor_l = _op.add(anchor_x, _op.multiply(anchor_w, minus_half))
+        anchor_r = _op.add(anchor_x, _op.multiply(anchor_w, plus_half))
+        anchor_t = _op.add(anchor_y, _op.multiply(anchor_h, minus_half))
+        anchor_b = _op.add(anchor_y, _op.multiply(anchor_h, plus_half))
+        anchor_expr = _op.concatenate([anchor_l, anchor_t, anchor_r, anchor_b], axis=1)
+        anchor_expr = _op.expand_dims(anchor_expr, 0)
+
+        # attributes for multibox_transform_loc
+        new_attrs0 = {}
+        new_attrs0["clip"] = False
+        new_attrs0["threshold"] = custom_options["nms_score_threshold"]
+        new_attrs0["variances"] = (
+            1/custom_options["x_scale"],
+            1/custom_options["y_scale"],
+            1/custom_options["w_scale"],
+            1/custom_options["h_scale"],
+        )
+
+        # attributes for non_max_suppression
+        new_attrs1 = {}
+        new_attrs1["return_indices"] = False
+        new_attrs1["iou_threshold"] = custom_options["nms_iou_threshold"]
+        new_attrs1["force_suppress"] = False
+        new_attrs1["top_k"] = anchor_boxes
+        new_attrs1["max_output_size"] = custom_options["max_detections"]
+        new_attrs1["invalid_to_bottom"] = False
+
+        ret = _op.vision.multibox_transform_loc(cls_pred, loc_prob,
+                                                anchor_expr, **new_attrs0)
+        ret = _op.vision.non_max_suppression(ret[0], ret[1], **new_attrs1)
```

Comment:
I'm curious about how much difference there exists between the nms logics of this op and standard tensorflow non_max_suppression op?

Reply:
They're not that different, but this op has some additional logic to determine the class of each box as well as do the transform from the box co-ords as generated by the convolutional network into screen space co-ords.